src/editor/json_simple_select.rs

```diff
@@ -0,0 +1,84 @@
+use anyhow::Context as _;
+
+pub fn apply(json: &str, selector: &str) -> anyhow::Result<String> {
+    let selector = selector
+        .split(['.', '[', ']'])
+        .filter(|part| !part.is_empty());
+    let value: serde_json::Value = serde_json::from_str(json)?;
+    let mut current = &value;
+    for part in selector {
+        match current {
+            serde_json::Value::Null => anyhow::bail!("can not select into a null value"),
+            serde_json::Value::Bool(_) => anyhow::bail!("can not select into a bool value"),
+            serde_json::Value::Number(_) => anyhow::bail!("can not select into a numeric value"),
+            serde_json::Value::String(_) => anyhow::bail!("can not select into a string value"),
+            serde_json::Value::Array(values) => {
+                let index = part
+                    .parse::<usize>()
+                    .context("selector can not be used to index an array")?;
+                current = values
+                    .get(index)
+                    .context("selector selected out of bounds in the array")?;
```

Comment:
```suggestion
                    .context(format!("selector selected out of bounds in the array, tried index {index} but array length was {:}", values.len()))?;
```
also a nit but with more intend: I could imagine that with complex enough json input, one might look at the wrong array by accident (user error). Getting reassurance which selector was used and how the array looked like might ease debugging. E.g.: 
```Rust
#[test]
fn simple_array_wrong() {
    case(r#"{"foo": [13, 37, 32, {"bar": 42}], "fooo": [99, {"bar": 77}]}"#, ".fooo[3].bar", "42");
}
```
yields: 
```
selector selected out of bounds in the array, tried index 3 but array length was 2
```